service/src/test/java/bio/terra/cbas/dependencies/bard/TestBardService.java

```diff
@@ -0,0 +1,173 @@
+package bio.terra.cbas.dependencies.bard;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import bio.terra.bard.api.DefaultApi;
+import bio.terra.bard.client.ApiClient;
+import bio.terra.bard.model.EventsEvent200Response;
+import bio.terra.bard.model.EventsEventLogRequest;
+import bio.terra.cbas.common.DateUtils;
+import bio.terra.cbas.dependencies.common.HealthCheck;
+import bio.terra.cbas.model.PostMethodRequest;
+import bio.terra.cbas.model.RunSetRequest;
+import bio.terra.cbas.model.WdsRecordSet;
+import bio.terra.cbas.models.GithubMethodDetails;
+import bio.terra.cbas.models.Method;
+import bio.terra.cbas.models.MethodVersion;
+import bio.terra.common.iam.BearerToken;
+import java.time.OffsetDateTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.web.client.RestClientException;
+
+@ExtendWith(MockitoExtension.class)
+class TestBardService {
+  private BardService bardService;
+  private DefaultApi defaultApi;
+  private BearerToken userToken;
+  private final String appId = "cbas";
+
+  @BeforeEach
+  void setup() {
+    BardClient bardClient = mock(BardClient.class);
+    bardService = new BardService(bardClient);
+    ApiClient apiClient = mock(ApiClient.class);
+    defaultApi = mock(DefaultApi.class);
+    userToken = new BearerToken("foo");
+    lenient().when(bardClient.apiClient()).thenReturn(apiClient);
+    lenient().when(bardClient.bardAuthClient(any())).thenReturn(apiClient);
```

Comment:
Might be nice to have two separate apiClients here so that you can verify that the right methods are being called on the right clients (eg `status` to the non-auth-client and `events` to the auth'd client)